### PR TITLE
refactor(ir): rewrite `ibis.expr.format` using `node.map()`

### PR DESCRIPTION
**Pull request by kszucs**
The goal here is to make the `format` implementation more "flat" and to make it more robust against newly defined operations (mostly relations). For now, this PR currently aims to reach feature parity with the previous implementation.

### CODE REVIEW

**Comment by kszucs** [comment]
In the future we could improve the representations by using rich renderables in various ways:
- color coding relation references
- syntax highlighting for node types 
- better table look and feel for inmemory tables and schemas

File: ibis/common/tests/test_graph.py [hunk 1]
```diff
@@ -178,3 +189,75 @@ class All(Bool):
...
     copied = node.copy(arguments=(T, F))
     assert copied == All((T, F), strict=False)
+
+
+class MySequence(Sequence):
+    def __init__(self, *items):
+        self.items = items
+
+    def __getitem__(self, index):
+        return self.items[index]
```
Comment (cpcloud) [inline ibis/common/tests/test_graph.py:199]: Are these missing test coverage?

File: ibis/expr/format.py [hunk 2]
```diff
@@ -1,684 +1,350 @@
...
+        return pieces
+
+    first_n = limit // 2
+    last_m = limit - first_n
+    first, last = pieces[:first_n], pieces[-last_m:]
+    if vertical:
+        maxlen = max(*map(len, first), *map(len, last))
+        ellipsis = util.VERTICAL_ELLIPSIS.center(maxlen)
+    else:
+        ellipsis = util.HORIZONTAL_ELLIPSIS
```
Comment (cpcloud) [inline ibis/expr/format.py:75]: Seems like a bunch of this function is missing test coverage.

Comment (kszucs) [reply ibis/expr/format.py:75]: Adding tests.

File: ibis/common/tests/test_graph.py (hunk 1 above)
Comment (kszucs) [reply ibis/common/tests/test_graph.py:199]: There are not supposed to be called, adding an assertion.

Comment (kszucs) [reply ibis/common/tests/test_graph.py]: Updated.

File: ibis/expr/format.py [hunk 3]
```diff
@@ -1,684 +1,344 @@
...
 import ibis.expr.types as ir
 from ibis import util
-from ibis.common import graph
 
-Aliases = Mapping[ops.TableNode, int]
-Deps = Deque[Tuple[int, ops.TableNode]]
+_infix_ops = {
+    # comparison operations
+    ops.Equals: "==",
+    ops.IdenticalTo: "===",
```
Comment (cpcloud) [inline ibis/expr/format.py:20]: CUTE

Comment (kszucs) [reply ibis/expr/format.py:20]: Cute?

**Review by cpcloud — APPROVED** [review]
SHIP IT!!

**Pull request merged**